### PR TITLE
added biogpt token classifier

### LINKED ISSUE #21786: BioGPT Token Classification
### Feature request

It would be nice to have this available.

### Motivation

working on biomedical token classification datasets and I would like to try BioGPT with them.

### Your contribution

I could send a PR if you want. I guess it shouldn't be too hard.

### PR DESCRIPTION
**Pull request by upjabir**
# What does this PR do?
Added Token Classifier for BioGpt

Fixes #21786

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@ArthurZucker @younesbelkada @NielsRogge @sgugger

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
Nice PR, thanks for adding this.  
It seems there is an issue with your CircleCI permissions, some of the tests won’t run.
Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)?

File: src/transformers/models/biogpt/modeling_biogpt.py [hunk 1]
```diff
@@ -731,3 +735,101 @@ def _reorder_cache(past_key_values, beam_idx):
...
+        else:
+            classifier_dropout = 0.1
+        self.dropout = nn.Dropout(classifier_dropout)
+        self.classifier = nn.Linear(config.hidden_size, config.num_labels)
+
+        self.post_init()
+
+        # Model parallel
+        self.model_parallel = False
+        self.device_map = None
```
Comment (younesbelkada) [inline src/transformers/models/biogpt/modeling_biogpt.py:766]:
```suggestion
```

File: src/transformers/models/biogpt/modeling_biogpt.py [hunk 2]
```diff
@@ -731,3 +735,101 @@ def _reorder_cache(past_key_values, beam_idx):
...
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.biogpt = BioGptModel(config)
+        if hasattr(config, "classifier_dropout") and config.classifier_dropout is not None:
+            classifier_dropout = config.classifier_dropout
+        elif hasattr(config, "hidden_dropout") and config.hidden_dropout is not None:
+            classifier_dropout = config.hidden_dropout
+        else:
+            classifier_dropout = 0.1
```
Comment (younesbelkada) [inline src/transformers/models/biogpt/modeling_biogpt.py:758]: nit: not sure if hardcoding default values like this is a good option

File: src/transformers/models/biogpt/modeling_biogpt.py (hunk 1 above)
Comment (younesbelkada) [reply src/transformers/models/biogpt/modeling_biogpt.py:766]: This can be removed

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot for this! LGTM with only 2 nits

File: src/transformers/models/biogpt/modeling_biogpt.py (hunk 1 above)
Comment (upjabir) [reply src/transformers/models/biogpt/modeling_biogpt.py:766]: okkay . Thanks for review

**Comment by upjabir** [comment]
@ArthurZucker am attaching the error from circleci in here. Please do a check on it.
```python
 python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      fatal: not a git repository (or any of the parent directories): .git
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-xnk_5lxf/onnx_fafa120a391e44089efd375a22a6dba2/setup.py", line 72, in <module>
          assert CMAKE, 'Could not find "cmake" executable!'
      AssertionError: Could not find "cmake" executable!
      [end of output]  

```

File: src/transformers/models/biogpt/modeling_biogpt.py [hunk 3]
```diff
@@ -731,3 +735,97 @@ def _reorder_cache(past_key_values, beam_idx):
...
```
Comment (younesbelkada) [inline src/transformers/models/biogpt/modeling_biogpt.py:758]:
```suggestion
        else:
            classifier_dropout = config.hidden_dropout_prob
```

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot for iterating! LGTM with only one comment

**Comment by ArthurZucker** [comment]
> @ArthurZucker am attaching the error from circleci in here. Please do a check on it.
> 
> ```python
>  python setup.py egg_info did not run successfully.
>   │ exit code: 1
>   ╰─> [7 lines of output]
>       fatal: not a git repository (or any of the parent directories): .git
>       Traceback (most recent call last):
>         File "<string>", line 2, in <module>
>         File "<pip-setuptools-caller>", line 34, in <module>
>         File "/tmp/pip-install-xnk_5lxf/onnx_fafa120a391e44089efd375a22a6dba2/setup.py", line 72, in <module>
>           assert CMAKE, 'Could not find "cmake" executable!'
>       AssertionError: Could not find "cmake" executable!
>       [end of output]  
> ```

You need to pull from main for this!

**Review by younesbelkada — APPROVED** [review]
Thanks for your contribution!

**Review by sgugger — APPROVED** [review]
Thanks for adding this model!

**Pull request merged**